sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.py

```diff
@@ -37,17 +37,17 @@ def produce_str() -> str:
 
 @func_to_container_op
 def produce_list_of_dicts() -> list:
-    return ({"aaa": "aaa1", "bbb": "bbb1"}, {"aaa": "aaa2", "bbb": "bbb2"})
+    return ([{"aaa": "aaa1", "bbb": "bbb1"}, {"aaa": "aaa2", "bbb": "bbb2"}],)
 
 
 @func_to_container_op
 def produce_list_of_strings() -> list:
-    return ("a", "z")
+    return (["a", "z"],)
 
 
 @func_to_container_op
 def produce_list_of_ints() -> list:
-    return (1234567890, 987654321)
+    return ([1234567890, 987654321],)
```

Comment:
this new syntax with the `[],` is breaking our current loop feature. it's required for all loops now? maybe we need to get rid of it in our compiler to avoid this issue.

Reply:
For now, let me revert these KFP 1.4.0 changes and create a separate issue. Though technically, if we claim KFP `1.4.0` alignment/support we may need those to work, no?

Reply:
It seems like a limitation from the Tekton parameter API. Feng has to remove this `[]` in order to pass the list argument to Tekton. Maybe we should create an issue on this to post process in our compiler.

Reply:
some of this work around is also apply to 1.3.0. We are still pending some Tekton api in order to run it like the Argo way as Tekton custom task has its own limitation. I can add some documents after this pr on what are the different between Tekton and Argo when running dynamic loops. Our previous loop implementation doesn't support dynamic loop so it's still an improvement.